docs/platform-engineers/addon/intro.md

```diff
@@ -100,175 +107,493 @@ metadata:
   name: velaux
   namespace: vela-system
 spec:
-# component definition of resource dir .
+  components:
+    - name: namespace
+      type: k8s-objects
+      properties:
+        objects:
+        - apiVersion: v1
+          kind: Namespace
+          metadata:
+            name: my-namespace
 ```
 
-You can describe anything in the application if you want, the rest resources defined in other folders will be automatically rendered and appended into this template as components.
+In this example, we define the basic framework of an application. This application contains a `k8s-objects` type component that contains `namespace` resource. After the addon is enabled, this namespace will be applied by KubeVela to all clusters.
 
-The name of Application in template will be replaced by the addon name in `metadata.yaml`. The addon application will always have a unified name in the format of `addon-<addon_name>`.
+#### template.cue
 
-> If the addon will be installed in managed clusters, please don't define policies or workflow steps in the template, or it will break the automatic precess built internally for addon installation.
+If you want the end user to determine some fields of the application by setting parameters when enabling the addon, you need write a `template.cue` file to do so. You can define the `template.cue` file like this to allow end-user specify the name of namespace in the above example:
 
-#### Examples
+```cue
+package main
+
+output: {
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
+	}
+	spec: {
+		components: [
+			{
+				type: "k8s-objects"
+				name: "example-namespace"
+				properties: objects: [{
+					apiVersion: "v1"
+					kind:       "Namespace"
+					metadata: name: parameter.namespace
+				}]
+			},
+		]
+	}
+}
+```
+
+As we can see, this CUE file mainly contain a `ouput` block. The `output` block must be a KubeVela application that may contain components, policies, or workflow. 
```

Comment:
The CUE file shown above contains an `output` field. The `output` field must embeds a standard KubeVela application, consisting components, policies, and workflow.